src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -161,7 +166,14 @@ private void OnHostingHttpRequestInStart(object arg)
                 string httpMethod = request.Method?.ToUpperInvariant() ?? "UNKNOWN";
                 string url = GetUrl(request);
 
-                string resourceUrl = UriHelpers.GetRelativeUrl(new Uri(url), tryRemoveIds: true)
+                string absolutePath = request.Path.Value;
+
+                if (request.PathBase.HasValue)
+                {
+                    absolutePath = request.PathBase.Value + absolutePath;
```

Comment:
This change looks correct to me, but should we add some tests to assert that the switch from what was basically
```csharp
new Uri(GetUrl(request)).AbsolutePath
```
to
```csharp
request.PathBase.Value + request.Path.Value
```
didn't change the resource name?

Reply:
You're right. Gonna introduce the test in a separate PR then rebase this one (otherwise we're not really testing)